### ISSUE TITLE
SQLSyntaxError on Update with nested @Embeddable 

### ISSUE THREAD

**Issue by jky0ung** [line/kotlin-jdsl#181]
### Entity with embeddable class name Address
```kotlin
@Entity
class Headquarter(
    var id: Long,
    var name: String,
    var ceoName: String,
    var address: Address,
    val isActive: Boolean = true
)
```
### Address with nested embeddable class name Coordinate
```kotlin
@Embeddable
data class Address(
    var address: String,
    var subAddress: String,
    @Embedded
    var coordinate: Coordinate
)
```
### Coordinate
```kotlin
@Embeddable
data class Coordinate(
    var longitude: Double,
    var latitude: Double
)
```
### Update JDSL Function
```kotlin
fun updateHeadquarter(headquarterId: Long, params: UpdateHeadquarterParams): Int {
    return queryFactory.updateQuery<Headquarter> {
        where(col(Headquarter::id).equal(headquarterId))
        params.name?.let { setParams(col(Headquarter::name) to it) }
        params.address?.let { setParams(col(Headquarter::address) to it )}
        params.ceoName?.let { setParams(col(Headquarter::ceoName) to it )}
    }.executeUpdate()
}
```
### Update Params Print
```
UpdateHeadquarterParams(name=test, ceoName=test, address=Address(address=서울 강남구 강남대로48길, subAddress=test, coordinate=Coordinate(longitude=127.034236192559, latitude=37.4868845435612)))
```
### Generated SQL
The problem is here. Generated SQL doesn't have longitude from Coordinate, and generated strange. 
```sql
2023-01-28 11:56:44.887 DEBUG 25463 --- [nio-2501-exec-8] org.hibernate.SQL                        : 
    update
        headquarter 
    set
        name=?,
        address=?=latitude=?=sub_address=?, <- HERE
        ceo_name=? 
    where
        id=1
Hibernate: 
    update
        headquarter 
    set
        name=?,
        address=?=latitude=?=sub_address=?, <- HERE
        ceo_name=? 
    where
        id=1
```
### Full Error Log
```sql
2023-01-28 11:56:44.887 DEBUG 25463 --- [nio-2501-exec-8] org.hibernate.SQL                        : 
    update
        headquarter 
    set
        name=?,
        address=?=latitude=?=sub_address=?,
        ceo_name=? 
    where
        id=5
Hibernate: 
    update
        headquarter 
    set
        name=?,
        address=?=latitude=?=sub_address=?,
        ceo_name=? 
    where
        id=5
2023-01-28 11:56:44.887 TRACE 25463 --- [nio-2501-exec-8] o.h.type.descriptor.sql.BasicBinder      : binding parameter [1] as [VARCHAR] - [test]
2023-01-28 11:56:44.887 TRACE 25463 --- [nio-2501-exec-8] o.h.type.descriptor.sql.BasicBinder      : binding parameter [2] as [VARCHAR] - [서울 강남구 강남대로48길]
2023-01-28 11:56:44.888 TRACE 25463 --- [nio-2501-exec-8] o.h.type.descriptor.sql.BasicBinder      : binding parameter [3] as [DOUBLE] - [37.4868845435612]
2023-01-28 11:56:44.888 TRACE 25463 --- [nio-2501-exec-8] o.h.type.descriptor.sql.BasicBinder      : binding parameter [4] as [DOUBLE] - [127.034236192559]
2023-01-28 11:56:44.888 TRACE 25463 --- [nio-2501-exec-8] o.h.type.descriptor.sql.BasicBinder      : binding parameter [5] as [VARCHAR] - [test]
2023-01-28 11:56:44.888 TRACE 25463 --- [nio-2501-exec-8] o.h.type.descriptor.sql.BasicBinder      : binding parameter [6] as [VARCHAR] - [test]
2023-01-28 11:56:44.890  WARN 25463 --- [nio-2501-exec-8] o.m.jdbc.message.server.ErrorPacket      : Error: 1292-22007: Truncated incorrect DOUBLE value: '서울 강남구 강남대로48길'
2023-01-28 11:56:44.890  WARN 25463 --- [nio-2501-exec-8] o.h.engine.jdbc.spi.SqlExceptionHelper   : SQL Error: 1292, SQLState: 22007
2023-01-28 11:56:44.890 ERROR 25463 --- [nio-2501-exec-8] o.h.engine.jdbc.spi.SqlExceptionHelper   : (conn=61) Truncated incorrect DOUBLE value: '서울 강남구 강남대로48길'
2023-01-28 11:56:44.891  WARN 25463 --- [nio-2501-exec-8] c.p.t.c.e.GlobalExceptionHandler         : could not execute statement; SQL [n/a]; nested exception is org.hibernate.exception.SQLGrammarException: could not execute statement
2023-01-28 11:56:44.894  WARN 25463 --- [nio-2501-exec-8] .m.m.a.ExceptionHandlerExceptionResolver : Resolved [org.springframework.dao.InvalidDataAccessResourceUsageException: could not execute statement; SQL [n/a]; nested exception is org.hibernate.exception.SQLGrammarException: could not execute statement]
```

#### Update with JPA Repository works fine
``` kotlin
fun updateHeadquarter(headquarterId: Long, params: UpdateHeadquarterParams): HeadquarterRes {
    params.run {
        val headquarter = headquarterJpaRepository.findById(headquarterId).get()
        address?.let {headquarter.address = it }
        name?.let { headquarter.name = it }
        ceoName?.let { headquarter.ceoName = it }
        headquarterJpaRepository.save(headquarter)
        return HeadquarterRes.fromEntity(headquarter)
    }
}
```

**Comment by cj848** [comment]
You need to use an associate.
It looks like there is no mapping.
associate(Headquater::address)
Try adding

**Comment by jky0ung** [comment]
> 

I tried this two things, and it's a same result.
```kotlin
associate(Headquarter::address)
```
```kotlin
associate(Headquarter::class, Address::class, on(Headquarter::address))
```

I think, this error occurs when there have a nested embedded class.
So, I tried this.
```kotlin
associate(Headquarter::class, Address::class, on(Headquarter::address))
associate(Address::class, Coordinate::class, on(Address::coordinate))
```

but same.

If I remove the child embedded class in the parent embedded class, it works well even if there is no associate!

**Comment by shouwn** [comment]
Hi @jky0ung 

I think this seems to be a bug in hibernate. And it seems to be fixed in the latest version of hibernate. I tried to test your example with the following code while developing Kotlin JDSL 3.0.0:

```kotlin
@Embeddable
data class EmployeeAddress(
    @Embedded
    val zipCode: ZipCode,

    @Column(name = "street_address_1")
    val streetAddress1: String,

    @Column(name = "street_address_2")
    val streetAddress2: String?,
)

@Embeddable
data class ZipCode(
    @Column(name = "zip_code")
    val zipCode: String,

    @Column(name = "city")
    val city: String,

    @Column(name = "province")
    val province: String,
)


employeeRepository.update {
    update(
        entity(Employee::class),
    ).set(
        path(Employee::address),
        EmployeeAddress(
            zipCode = ZipCode(
                zipCode = "1",
                city = "1",
                province = "1",
            ),
            streetAddress1 = "1",
            streetAddress2 = "1",
        ),
    ).where(
        path(Employee::employeeId).equal(1L),
    )
}

employeeRepository.findAll {
    select(
        path(Employee::address),
    ).from(
        entity(Employee::class),
    )
}.forEach {
    println(it)
}
```

I apologize for the long wait.... I saw this issue too late and was slow to respond. If you are still using Kotlin JDSL, please check it out. 

Also Kotlin JDSL 3.0.0-SNAPSHOT has been released in snapshot repository. If you don't mind, please try it with Kotlin JDSL 3.0.0-SNAPSHOT.

The following docs will help you get started with Kotlin JDSL 3.0.0.
https://kotlin-jdsl.gitbook.io/docs/

**Comment by shouwn** [comment]
I am closing this issue for now. If the issue is not resolved in 3.0.0, please reopen it.
